embedding-store-filter-parsers/langchain4j-embedding-store-filter-builder-language/src/main/java/dev/langchain4j/store/embedding/filter/builder/language/TableDefinition.java

```diff
@@ -0,0 +1,294 @@
+package dev.langchain4j.store.embedding.filter.builder.language;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Defines the structure of metadata fields that can be used in filters.
+ * 
+ * <p>A TableDefinition describes the available metadata columns, their types,
+ * and descriptions that help the language model understand how to create
+ * appropriate filters. This is essential for converting natural language
+ * queries into structured filter operations.
+ * 
+ * <p>Example usage:
+ * <pre>{@code
+ * TableDefinition tableDefinition = TableDefinition.builder()
+ *     .addColumn("author", String.class, "The document author")
+ *     .addColumn("publishDate", java.time.LocalDate.class, "Publication date")
+ *     .addColumn("rating", Number.class, "User rating from 1-5")
+ *     .addColumn("category", CategoryEnum.class, "Document category")
+ *     .build();
+ * }</pre>
+ * 
+ * <p>The class supports various data types including:
+ * <ul>
+ *   <li>Primitive types: String, Number, Boolean</li>
+ *   <li>Date types: LocalDate, LocalDateTime</li>
+ *   <li>Enum types for categorical data</li>
+ * </ul>
+ * 
+ * @since 1.0.0
+ */
+public class TableDefinition {
+
+    private final List<ColumnDefinition> columns;
+
+    /**
+     * Constructs a new TableDefinition with the specified columns.
+     * 
+     * @param columns the list of column definitions. Must not be null.
+     * @throws NullPointerException if columns is null
+     */
+    public TableDefinition(List<ColumnDefinition> columns) {
+        this.columns = Objects.requireNonNull(columns, "columns cannot be null");
+    }
+
+    /**
+     * Returns the list of column definitions for this table.
+     * 
+     * @return an unmodifiable list of column definitions
+     */
+    public List<ColumnDefinition> getColumns() {
```

Comment:
```suggestion
    public List<ColumnDefinition> columns() {
```
Please check everywhere that there is no "get" in getter names